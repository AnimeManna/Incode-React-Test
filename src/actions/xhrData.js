import {dataClients} from '../api/dataClients'

import {
    XHR_CLIENTS_START,
    XHR_CLIENTS_SUCCESS,
    XHR_CLIENTS_ERROR
} from '../ActionsType/ActionTypes'

export const xhrData = () => async dispatch => {
    dispatch({type:XHR_CLIENTS_START});
    try{
        const clients = await dataClients();
        clients.map((value,key)=>{
            value.id=key+1;        // добавляем id каждому пользователю
            return{
                ...value,
                ...value.id
            }
        });
        dispatch({type:XHR_CLIENTS_SUCCESS, payload:clients})

    }catch (error){
        dispatch({
            type:XHR_CLIENTS_ERROR,
            payload:error,
            error:true
        })

    }
}