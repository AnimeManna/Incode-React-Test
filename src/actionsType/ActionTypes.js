export const XHR_CLIENTS_START = 'XHR_CLIENTS_START'
export const XHR_CLIENTS_SUCCESS = 'XHR_CLIENTS_SUCCESS'
export const XHR_CLIENTS_ERROR = 'XHR_CLIENTS_ERROR'