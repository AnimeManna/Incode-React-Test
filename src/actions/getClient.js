import {GET_CLIENT} from '../ActionsType/ActionTypes'

export const getClient =  (id) => dispatch => {
    console.log('ID',id);
    dispatch({type:GET_CLIENT,payload:id})
}