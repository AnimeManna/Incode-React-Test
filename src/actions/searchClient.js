import {SEARCH_CLIENT} from "../actionsType/ActionTypes";

export const searchClient = (search) => dispatch =>{
    console.log(search)
    dispatch({type:SEARCH_CLIENT,payload:search})
}