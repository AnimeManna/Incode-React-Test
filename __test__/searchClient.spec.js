import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

import {searchClient} from '../src/actions/searchClient'

import {SEARCH_CLIENT} from "../src/actionsType/ActionTypes";

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

describe('searchClint_actions', ()=>{

    const initialState = {
    }
    const store = mockStore(initialState)

    test('dispatch searchClient',()=>{
        store.dispatch(searchClient('test'));

        const actions = store.getActions();
        expect(actions).toMatchSnapshot();
    })


})