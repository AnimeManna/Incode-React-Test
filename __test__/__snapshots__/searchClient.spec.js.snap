// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchClint_actions dispatch searchClient 1`] = `
Array [
  Object {
    "payload": "test",
    "type": "SEARCH_CLIENT",
  },
]
`;
